### PR TITLE
Type definition for Bootstrap3-Dialog version 1.35.4

### PR DESCRIPTION
**Pull request by CNBoland**
Please fill in this template.

- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ x ] The package does not provide its own types, and you can not add them.
- [ x ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ x ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ x ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/bootstrap3-dialog/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
+// Type definitions for bootstrap3-dialog 1.35
```
Comment (armanio123) [inline types/bootstrap3-dialog/index.d.ts]: I'd suggest enclosing these interfaces on their namespace with the same name as your variable declaration: BootstrapDialog

File: types/bootstrap3-dialog/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    getModalDialog(): JQuery;
+    /** Return the raw modal header. */
+    getModalHeader(): JQuery;
+    /** Return the raw modal footer. */
+    getModalFooter(): JQuery;
+    /** Return the raw modal body. */
+    getModalBody(): JQuery;
+}
+
+declare var BootstrapDialog: BootstrapDialogStatic;
```
Comment (armanio123) [inline types/bootstrap3-dialog/index.d.ts]: Use const instead of var.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/bootstrap3-dialog/index.d.ts (hunk 1 above)
Comment (CNBoland) [reply types/bootstrap3-dialog/index.d.ts]: I like the namespace idea. I could then shorten some of the type names from `BootstrapDialog.BootstrapDialogContext` to `BootstrapDialog.DialogContext`.

What do you think?

File: types/bootstrap3-dialog/index.d.ts (hunk 2 above)
Comment (CNBoland) [reply types/bootstrap3-dialog/index.d.ts]: Agreed

File: types/bootstrap3-dialog/index.d.ts (hunk 1 above)
Comment (armanio123) [reply types/bootstrap3-dialog/index.d.ts]: I think it's the best as well.

**Review by armanio123 — APPROVED** [review]

**Pull request merged**